feat(swagger): Springfox to openapi migration

MichaelsJP opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Close #1027 

### Information about the changes
- Key functionality added: Upgrade from springfox to springboot openapi 
- Reason for change: The springfox package is not maintained and outdated. It's also not compatible with spring-boot 3, which is the only blocker currently to not upgrade.

### Required changes to ors config (if applicable)
- Added `swagger_documentation_add_autogenerated_url` to let the user decide if the auto-generated (localhost) url should be added to the available servers. This is needed since the openapi package defines the servers as an array of possible endpoints, rather than a fixed string for one url only.

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

TheGreatRefrigerator commented on openrouteservice/src/main/java/org/heigit/ors/api/OpenApiConfiguration.java:
```diff
@@ -0,0 +1,104 @@
+    private Info apiInfo() {
+        return new Info()
+                .title(SERVICE_NAME)
+                .description("This is the openrouteservice API documentation")
```
If it's possible somehow it would be great to have the ors version tag dynamically provided here.
We currently do this in the api-doc service:
https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api/-/blob/main/cron/docs_update_check.py#L48-65

MichaelsJP replied:
> ors_live_version = version.parse(status.get('engine').get('version'))

You're referring to the ors engine version?
As I understand the version tag in the `apiInfo` object, it's targeted towards the API version (v1/v2/v3).

As stated in the [swagger docs](https://swagger.io/docs/specification/api-general-info/) we could use a versioning approach. E.g. `version: v2-7.0.1` or vice versa.

That way we could always extract the ors version from the docs and also know with what ors version they were generated.

MichaelsJP replied:
We could dynamically adjust the description part with the normal version.
An edge case would be if ors is run from master (nightly). -> This also applies to /status in effect, so we could also forget about it ;).

MichaelsJP commented on openrouteservice/src/main/java/org/heigit/ors/api/OpenApiConfiguration.java:
```diff
@@ -0,0 +1,106 @@
+        return new Info()
+                .title(SERVICE_NAME)
+                .description("This is the openrouteservice API V2 documentation for ORS Core-Version " + VERSION  + ". Documentations for [older Core-Versions](https://github.com/GIScience/openrouteservice-docs/releases) can be rendered with the [Swagger-Editor](https://editor-next.swagger.io/).")
+                .version("2")
```
```suggestion
                .version("v2")
```

jarinox commented:
There is a `ptRequest` field generated for `directions/*` POST routes which should not be there.  

![image](https://github.com/GIScience/openrouteservice/assets/45308098/bf862a23-8776-4e23-88a0-7daec5e2ee27)

MichaelsJP commented:
@jarinox Yes and no. We'll have public transit soon in our live API and the code supports it already. A curl request looks like this

```
curl -d '{"coordinates": [[8.6729581, 49.4468535], [8.7067204, 49.3786147]], "departure": "2022-07-04T13:02:26Z"}'` \
-H "Content-Type: application/json" -X POST http://localhost:8080/ors/v2/directions/pt
````
Makes sense with the parameter, right?
But maybe it could be worth hiding it from our live API docs until it's really available.

MichaelsJP committed:
feat(swagger): Upgrade springfox to openapi

This removes the outdated springfox library that was serving the swagger <3 version.
openapi introduces swagger with version >=3 using the new openapi standard. The package is maintained and compatible with newer spring-boot versions.

Co-authored-by: Amandus Butzer <amandus.butzer@heigit.org>

takb reviewed (approved):
finally! lgtm.
